client/src/components/WriteMessage.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { connect } from 'react-redux';
+import { Segment, Button, TextArea} from 'semantic-ui-react';
+import io from 'socket.io-client';
+import * as ChatActions from './../actions/chatActionCreator.jsx';
+
+class WriteMessage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: '',
+    }
+    this.handleFormChange = this.handleFormChange.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.retrieveMessages = this.retrieveMessages.bind(this);
+  }
+
+  handleFormChange(e) {
+    this.setState({
+      text: e.target.value
+    })
+  }
+
+  handleFormSubmit() {
+    var messagePayload = {
+      text: this.state.text,
+      sender_id: this.props.userId,
+      receiver_id: this.props.receiverId,
+      roomname: this.props.roomname,
+    }
+    this.setState({
+      text: '',
+    })
+    fetch('/messages/', {
+      method: 'POST',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+      }),
+      body: JSON.stringify(messagePayload)
+    })
+    .then((response) => {
+      if (!response.ok) {
+        throw Error('message failed to send...');
+      }
+      return response.json();
+    })
+    .then((data) => {
+      console.log('message posted to DB! data: ', data);
+      let {dispatch} = this.props;
+      dispatch(ChatActions.chatMessage(messagePayload));
+    })
+    .catch((error) => {
+      console.log('handleFormSubmit failed! Error: ', error);
+    })
+  }
+
+  retrieveMessages() {
+    fetch(`/messages/${Number(this.props.userId)}/?receiver_id=${Number(this.props.receiverId)}`, {
+      method: 'GET',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${localStorage.getItem('authToken')}`,
+      })
+    })
+    .then((response) => {
+      if (!response.ok) {
+        throw Error('failed to retrieve messages...')
+      }
+      return response.json();
+    })
+    .then((data) => {
+      console.log('messages retrieved! data: ', data);
+    })
+    .catch((error) => {
+      console.log('retrieveMessages failed! error: ', error);
+    })
+  }
+
+  componentDidMount() {
+    setTimeout(this.retrieveMessages, 1500);
```

Comment:
because the app doesnt hold onto the receiver data between site visits, retrieve messages is fairly pointless for the time being. I left it in there because I'm assuming we'll want it to be implemented later